fontbe/src/glyphs.rs

```diff
@@ -0,0 +1,280 @@
+
+        match glyph {
+            CheckedGlyph::Composite { name, .. } => {
+                error!("setting no glyph for {name}; composites not implemented yet");
+            }
+            CheckedGlyph::Contour { name, contours } => {
+                // Draw the default outline of our simple glyph
+                let Some(path) = contours.get(default_location) else {
+                    return Err(Error::GlyphError(ir_glyph.name.clone(), GlyphProblem::MissingDefault));
+                };
+                let base_glyph = SimpleGlyph::from_kurbo(path).map_err(|e| Error::KurboError {
+                    glyph_name: self.glyph_name.clone(),
+                    kurbo_problem: e,
+                    context: path.to_svg(),
+                })?;
+                context.set_glyph(name, base_glyph.into());
+            }
+        }
+
+        Ok(())
+    }
+}
+
+fn cubics_to_quadratics(glyph: CheckedGlyph) -> CheckedGlyph {
+    let CheckedGlyph::Contour { name, contours } = glyph else {
+        return glyph;  // nop for composite
+    };
+
+    trace!("Convert '{name}' to quadratic");
+
+    // put all the loc + path iters into a vec
+    let mut loc_iters: Vec<_> = contours
+        .iter()
+        .map(|(loc, path)| (loc, path.iter()))
+        .collect();
+    let mut new_contours = HashMap::<NormalizedLocation, BezPath>::new();
+
+    let mut subpath_start_pts: Vec<_> = loc_iters
+        .iter_mut()
+        .filter_map(|(loc, iter)| match iter.next() {
+            Some(PathEl::MoveTo(p)) => {
+                new_contours.entry((*loc).clone()).or_default().move_to(p);
+                Some(p)
+            }
+            None => None,
+            Some(other) => panic!("'{name}': illegal start of path: {other:?}"),
+        })
+        .collect();
+    let mut prev_el_end_pts = subpath_start_pts.clone();
+    loop {
+        let elements: Vec<_> = loc_iters
+            .iter_mut()
+            .filter_map(|(_, iter)| iter.next())
+            .collect();
+        if elements.is_empty() {
+            break; // normal termination, we exhausted all the element iterators
+        }
+
+        // No work to be done if the segment isn't cubic
+        if let PathEl::CurveTo(..) = elements[0] {
+            if prev_el_end_pts.len() != elements.len() {
+                panic!("'{name}': not enough end points");
+            }
+            let cubics: Vec<_> = elements
+                .iter()
+                .zip(&prev_el_end_pts)
+                .map(|(el, p0)| match el {
+                    PathEl::CurveTo(p1, p2, p3) => CubicBez {
+                        p0: *p0,
+                        p1: *p1,
+                        p2: *p2,
+                        p3: *p3,
+                    },
+                    _ => unreachable!("'{name}': *must* all be cubic"),
+                })
+                .collect();
+
+            // At long last, actually convert something to quadratic
+            // TODO what should we pass for accuracy
```

Comment:
python cu2qu defaults to a _relative_ max_error of 1/1000th of the font's UPEM; e.g. for a usual font with head.unitsPerEm=1000, the _absolute_ max_error is 1 font unit.